detectors/gradle/gradle.go

```diff
@@ -159,7 +154,7 @@ type gradleProjectRootEntry struct {
 }
 
 func detectGradleProjectRoot(searchDir direntry.DirEntry) (*gradleProjectRootEntry, error) {
-	gradlewFileEntry := searchDir.FindFirstEntryByName("gradlew", false)
+	gradlewFileEntry := searchDir.FindImmediateChildByName("gradlew", false)
```

Comment:
With this change, the Gradle detector searches for the Gradle project in the provided directory (projectRootDirEntry), instead of creating a project based on the first Gradle wrapper script. This support scanning repos with multiple Gradle projects.